apps/web/src/pages/templates/workflow/workflow/node-types/WorkflowNode.tsx

```diff
@@ -102,24 +104,50 @@ export function WorkflowNode({
   const { isLimitReached: isSmsLimitReached } = useIntegrationLimit(ChannelTypeEnum.SMS);
   const [hover, setHover] = useState(false);
   const isMultiProviderConfigurationEnabled = useIsMultiProviderConfigurationEnabled();
-
-  const hasActiveIntegration = useMemo(() => {
-    const isChannelStep = [StepTypeEnum.EMAIL, StepTypeEnum.PUSH, StepTypeEnum.SMS, StepTypeEnum.CHAT].includes(
+  const { colorScheme } = useMantineColorScheme();
+  const isChannelStep = useMemo(() => {
+    return [StepTypeEnum.IN_APP, StepTypeEnum.EMAIL, StepTypeEnum.PUSH, StepTypeEnum.SMS, StepTypeEnum.CHAT].includes(
       channelType
     );
+  }, [channelType]);
+
+  const integrationsByEnv = useMemo(() => {
+    return integrations?.filter((integration) => integration._environmentId === environment?._id);
+  }, [environment, integrations]);
+
+  const hasActiveIntegration = useMemo(() => {
     const isEmailStep = channelType === StepTypeEnum.EMAIL;
     const isSmsStep = channelType === StepTypeEnum.SMS;
 
     if (isChannelStep) {
-      const isActive = !!integrations?.some((integration) => integration.channel === tabKey);
+      const isActive = !!integrationsByEnv?.some((integration) => integration.channel === tabKey);
       const isEmailStepActive = isEmailStep && !isEmailLimitReached;
       const isSmsStepActive = isSmsStep && !isSmsLimitReached;
 
       return isActive || isEmailStepActive || isSmsStepActive;
     }
 
     return true;
-  }, [integrations, tabKey, isEmailLimitReached, isSmsLimitReached]);
+  }, [integrationsByEnv, tabKey, isEmailLimitReached, isSmsLimitReached, isChannelStep]);
+
+  const getPrimaryIntegration = useMemo(() => {
+    if (!hasActiveIntegration) {
+      return undefined;
+    }
+    if (channelType === StepTypeEnum.IN_APP) {
+      return InAppProviderIdEnum.Novu;
+    }
```

Comment:
this if statement is not needed as it will fall into `if (isChannel)` and second return case